tests/test_cli.py

```diff
@@ -0,0 +1,224 @@
+import json
+import os
+from pathlib import Path
+from unittest.mock import MagicMock, Mock, patch
+from uuid import uuid4
+
+import pytest
+import yaml
+from click.testing import CliRunner
+from sagemaker.workflow.pipeline import Pipeline as SageMakerPipeline
+
+from kedro_sagemaker import cli
+from kedro_sagemaker.config import KedroSageMakerPluginConfig
+from kedro_sagemaker.constants import (
+    KEDRO_SAGEMAKER_ARGS,
+    KEDRO_SAGEMAKER_PARAM_KEY_PREFIX,
+    KEDRO_SAGEMAKER_PARAM_VALUE_PREFIX,
+    KEDRO_SAGEMAKER_WORKING_DIRECTORY,
+)
+from kedro_sagemaker.generator import KedroSageMakerGenerator
+from tests.utils import assert_has_any_call_with_args
+
+
+@patch("click.confirm", return_value=True)
+@pytest.mark.parametrize(
+    "clean_dir", (False, True), ids=("with existing config", "empty dir")
+)
+@pytest.mark.parametrize("yes", (False, True), ids=("without --yes", "with --yes"))
+def test_can_initialize_basic_plugin_config(
+    click_confirm,
+    clean_dir,
+    yes,
+    patched_kedro_package,
+    cli_context,
+):
+    # given
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        Path("conf/base").mkdir(parents=True)
+
+        if not clean_dir:
+            (Path.cwd() / "Dockerfile").write_text("FROM python:3.10")
+            (Path.cwd() / ".dockerignore").write_text("")
+
+        # when
+        result = runner.invoke(
+            cli.init,
+            [
+                "__bucket_name__",
+                "__execution_role__",
+                "__docker_image__",
+            ]
+            + (["--yes"] if yes else []),
+            obj=cli_context,
+        )
+
+        # then
+        assert result.exit_code == 0
+
+        config_path = Path("conf/base/sagemaker.yml")
+        assert config_path.exists()
+        config = KedroSageMakerPluginConfig.parse_obj(
+            yaml.safe_load(config_path.read_text())
+        )
+        assert config.aws.execution_role == "__execution_role__"
+        assert config.aws.bucket == "__bucket_name__"
+        assert config.docker.image == "__docker_image__"
+
+        if not yes and not clean_dir:
+            click_confirm.assert_called()
+        else:
+            click_confirm.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    "extra_params",
+    ["", json.dumps({"my_extra_param": 123, "my": {"nested": {"parameter": 66.6}}})],
+    ids=("without extra params", "with extra params"),
+)
+def test_can_compile_the_pipeline(
+    extra_params, patched_kedro_package, cli_context, dummy_pipeline, tmp_path: Path
+):
+    runner = CliRunner()
+    with patch.object(
+        KedroSageMakerGenerator, "get_kedro_pipeline", return_value=dummy_pipeline
```

Comment:
It's perfectly fine, but in the future you can consider getting rid of this `patch.object` and patch the context instead:

```
@patch("kedro.framework.project.pipelines", {"__default__": dummy_pipeline})
```